UT4MasterServer/Controllers/UnrealTournamentGameController.cs

```diff
@@ -111,9 +114,66 @@ public UnrealTournamentGameController(ILogger<SessionController> logger, Account
 			return Json(obj);
 		}
 
-		[HttpGet("profile/{id}/client/SetAvatarAndFlag")]
-		public IActionResult SetAvatarAndFlag(string id, [FromQuery] string profileId, [FromQuery] string rvn, [FromBody] string body)
+		[HttpPost("profile/{id}/client/SetAvatarAndFlag")]
+		public async Task<IActionResult> SetAvatarAndFlag(string id, [FromQuery] string profileId, [FromQuery] string rvn)
 		{
+			if (User.Identity is not EpicUserIdentity user)
+				return Unauthorized();
+
+			if (user.Session.AccountID != EpicID.FromString(id))
+				return Unauthorized();
```

Comment:
Move it into `JsonAPIController` as helper checks

Reply:
Hmm, yes that is a good idea.